SPHINXsys/src/shared/geometries/level_set.h

```diff
@@ -72,104 +69,111 @@ namespace SPH
 	class LevelSet
 		: public MeshWithDataPackages<BaseLevelSet, LevelSetDataPackage>
 	{
 	public:
-		ConcurrentVector<LevelSetDataPackage*> core_data_pkgs_; /**< packages near to zero level set. */
+		ConcurrentVector<LevelSetDataPackage *> core_data_pkgs_; /**< packages near to zero level set. */
 		Real global_h_ratio_;
 
-		LevelSet(BoundingBox tentative_bounds, Real data_spacing, 
-			ComplexShape& complex_shape, ParticleAdaptation& particle_adaptation);
-		virtual ~LevelSet() {};
+		LevelSet(BoundingBox tentative_bounds, Real data_spacing,
+				 ComplexShape &complex_shape, ParticleAdaptation &particle_adaptation);
+		virtual ~LevelSet(){};
 
-		virtual Real probeSignedDistance(const Vecd& position) override;
-		virtual Vecd probeNormalDirection(const Vecd& position) override;
-		virtual Real probeKernelIntegral(const Vecd& position, Real h_ratio = 1.0) override;
-		virtual Vecd probeKernelGradientIntegral(const Vecd& position, Real h_ratio = 1.0) override;
+		virtual bool probeIsWithinMeshBound(const Vecd &position) override;
+		virtual Real probeSignedDistance(const Vecd &position) override;
+		virtual Vecd probeNormalDirection(const Vecd &position) override;
+		virtual Real probeKernelIntegral(const Vecd &position, Real h_ratio = 1.0) override;
+		virtual Vecd probeKernelGradientIntegral(const Vecd &position, Real h_ratio = 1.0) override;
 		virtual void cleanInterface(bool isSmoothed = false) override;
-		virtual void writeMeshToPltFile(std::ofstream& output_file) override;
+		virtual void writeMeshFieldToPlt(std::ofstream &output_file) override;
 		bool isWithinCorePackage(Vecd position);
-		Real computeKernelIntegral(const Vecd& position);
-		Vecd computeKernelGradientIntegral(const Vecd& position);
+		Real computeKernelIntegral(const Vecd &position);
+		Vecd computeKernelGradientIntegral(const Vecd &position);
+
 	protected:
-		Kernel& kernel_;
+		Kernel &kernel_;
 
-		Real computeHeaviside(Real phi, Real half_width);
 		void reinitializeLevelSet();
 		void markNearInterface();
 		void redistanceInterface();
 		void updateNormalDirection();
-		void updateNormalDirectionForAPackage(LevelSetDataPackage* inner_data_pkg, Real dt = 0.0);
+		void updateNormalDirectionForAPackage(LevelSetDataPackage *inner_data_pkg, Real dt = 0.0);
 		void updateKernelIntegrals();
-		void updateKernelIntegralsForAPackage(LevelSetDataPackage* inner_data_pkg, Real dt = 0.0);
-		void stepReinitializationForAPackage(LevelSetDataPackage* inner_data_pkg, Real dt = 0.0);
-		void markNearInterfaceForAPackage(LevelSetDataPackage* core_data_pkg, Real dt = 0.0);
-		void redistanceInterfaceForAPackage(LevelSetDataPackage* core_data_pkg, Real dt = 0.0);
-		virtual void initializeDataInACell(Vecu cell_index, Real dt) override;
-		virtual void initializeAddressesInACell(Vecu cell_index, Real dt) override;
-		virtual void tagACellIsInnerPackage(Vecu cell_index, Real dt) override;
+		void updateKernelIntegralsForAPackage(LevelSetDataPackage *inner_data_pkg, Real dt = 0.0);
```

Comment:
also here

Reply:
The data Package is computing intensive.